### PR TITLE
Joey Turret!

### PR DESCRIPTION
**Pull request by ChaseBENNC9**
## Describe your changes

Implementing the joey turret enemy ability. This will happen at random times during the match with the kangaroo enemy type. The Turret can turn to face players but will be stationary.
Each enemy can place one turret and must wait until it expires before they can place another one. (Max of 4 active turrets) 

- [x] Spawning Turret 
- [x] Turret Can throw snowballs
- [x] Turret has a timeout 
- [x] Ability is randomly used 
- [x] Ability has a cooldown before it can be used again
- [x] Turret tracks player while turning

## Issue number and link

#35 #21

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by ChaseBENNC9** [comment]
Ready to Review
Scene is setup in JoeyTurret
Each enemy has a x% chance of placing a turret and the end of their path. you can modify the chance with the slider in the inspector.
Joey will last for a set ammount of time, at this moment its 15 seconds after which it will be destroyed
the joey firing rate is supposed to be faster but it might need to change to feel right
for joey turning i have two options.
a) It fires at the set angles like the main enemy 
b) it will constantly track and face towards the closest player.

to see each option there is a commented out line 
`if (CheckForPlayerDirection() != Vector3.zero) //If the player is in a direction`
found at `JoeyTurret.cs` line 33
that can be used to switch between them. that line will make it only use set directions (a) or if commented out (b) will be used. 

I've also just fixed the bug where friendly fire can happen as i needed a fix in this branch anyway

**Review by PBWiseman — CHANGES_REQUESTED** [review]
# Code Review

## Functional Requirements
_Is the application fit for purpose? Does it meet the user’s requirements and properly address the issue?_  

 - Good feature. Placement and shooting work well.

## Correctness of the Code
_Is the code reliable, giving same results every time it runs? Does it function correctly?_  

 - The Joeys can sometimes spawn inside and beyond the walls of the play arena. Obviously these walls are temporary but there will be both outer bounds that we don't want them spawning in as well as walls within the play space
image

- The Joey currently has no collision. Do we want there to be collision?

- All Kangaroos other than the first one currently have the turret placing script disabled and the % chance set to zero. I assume this was intentional for testing but don't forget to change that (and remove the debug.log) before merging.

# Final Comments

 - Happy for this to be merged when bugs addressed!

***

File: Assets/Scripts/KangarooAbility.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            activeTurret = joeyTurret;
+            canUseTurret = false;
+
+         } 
+    }
+
+
+    void Update()
+    {
+        Debug.Log("canUSeTurret: "  + canUseTurret);
```
Comment (erokasyuart) [inline Assets/Scripts/KangarooAbility.cs:52]: Make sure this gets removed

**Review by erokasyuart — CHANGES_REQUESTED** [review]
# Code Review

## Functional Requirements
_Is the application fit for purpose? Does it meet the user’s requirements and properly address the issue?_  

 - Yes, Joeys spawn and shoot.


## Legibility and Style
_Uses agreed coding standards/conventions? Readable, consistent and well-structured with useful comments?_  

 - (EnterTextHere)

_Any repetitive code that could be refactored? (DRY)_  

 - No issue


## Correctness of the Code
_Is the code reliable, giving same results every time it runs? Does it function correctly?_  

 - Good


## Implementation
_Any notable performance/speed issues? Compatibility and Responsiveness on other platforms ?_  

 - No


# Final Comments

 - The joey turret continues to shoot during the game end screen
***

**Comment by PBWiseman** [comment]
Issues addressed!

**Review by PBWiseman — APPROVED** [review]

**Review by erokasyuart — APPROVED** [review]

**Pull request merged**